network/conduit.go

```diff
@@ -19,6 +20,32 @@ func (c Channel) String() string {
 	return string(c)
 }
 
+// Len returns length of the ChannelList in the number of stored Channels.
+// It satisfies the sort.Interface making the ChannelList sortable.
+func (cl ChannelList) Len() int {
+	return len(cl)
+}
+
+// Less returns true if element i in the ChannelList  is less than j based on the numerical value of its Channel.
+// Otherwise it returns true.
+// It satisfies the sort.Interface making the ChannelList sortable.
+func (cl ChannelList) Less(i, j int) bool {
+	return cl[i] < cl[j]
+}
+
+// Swap swaps the element i and j in the ChannelList.
+// It satisfies the sort.Interface making the ChannelList sortable.
+func (cl ChannelList) Swap(i, j int) {
+	cl[i], cl[j] = cl[j], cl[i]
+}
+
+// ID returns hash of the content of ChannelList. It first sorts the ChannelList and then takes its
+// hash value.
+func (cl ChannelList) ID() flow.Identifier {
+	sort.Sort(cl)
+	return flow.MakeID(cl)
+}
```

Comment:
`SubscriptionManager` keeps `ChannelList` as a key-value map. Hence, when it extracts the channels list from the _map_, there will be no guarantee on the order of channels on the list. To avoid inconsistent identifier for the same channels list, we sort the channel list first, and then make the identifier.